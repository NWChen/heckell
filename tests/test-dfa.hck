let makeDFA : ((int set) * (char set) * (int * char -> int map) * int * (int set) * (char array)) -> (string -> bool);
makeDFA(Q, A, t, q0, F, w) = 
	let s : int;
	s = q0;
	for c in w do 
		print("\(c)");
	end	
	s in F
end

let Q = {1,2,3};
let A = {'0', '1'};

let t : int * char -> int map;
t = {((1, '0'), 1), ((1, '1'), 2), 
	 ((2, '0'), 1), ((2, '1'), 3),
	 ((3, '0'), 1), ((3, '1'), 3)};

let q0 : int;
q0 = 1;
let F : int set;
F = {3};
let w : char array;
w = ['0', '1', '0', '1', '1'];
let DFA : bool;
DFA = makeDFA(Q, A, t, q0, F, w);
print("Does '11' appear in \(w)? \(DFA)");