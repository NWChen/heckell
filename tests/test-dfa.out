0
1
0
1
1
Does '11' appear in w? true
